api/enki.py

```diff
@@ -0,0 +1,633 @@
+        medium=Edition.BOOK_MEDIUM,
+        #series=series,
+        publisher=element["publisher"],
+        #imprint=imprint,
+        #published=publication_date,
+        primary_identifier=primary_identifier,
+        identifiers=identifiers,
+        #subjects=subjects,
+        contributors=contributors,
+        )
+        #TODO: This should parse the content type and look it up in the Enki Delivery Data above. Currently,
+        # we assume everything is an ePub that uses Adobe DRM, which is a safe assumption only for now.
+        formats = []
+        formats.append(FormatData(content_type=Representation.EPUB_MEDIA_TYPE, drm_scheme=DeliveryMechanism.ADOBE_DRM))
+
+        circulationdata = CirculationData(
+            data_source=DataSource.ENKI,
+            primary_identifier=primary_identifier,
+            formats=formats,
+        )
+
+        metadata.circulation = circulationdata
+        return metadata
+
+
+    def process_one(self, element, ns):
+        if self.include_bibliographic:
+            bibliographic = self.extract_bibliographic(element, ns)
+        else:
+            bibliographic = None
+
+        passed_availability = None
+        if bibliographic and bibliographic.circulation:
+            passed_availability = bibliographic.circulation
+
+        if self.include_availability:
+            availability = self.extract_availability(circulation_data=passed_availability, element=element, ns=ns)
+        else:
+            availability = None
+
+        return bibliographic, availability
+
+class EnkiImport(Monitor):
+    """Maintain LicensePools for Enki titles.
+    """
+
+    VERY_LONG_AGO = datetime.datetime(1970, 1, 1)
+    FIVE_MINUTES = datetime.timedelta(minutes=5)
+
+    def __init__(self, _db, name="Enki Import",
+                 interval_seconds=60, batch_size=50, api=None):
+	super(EnkiImport, self).__init__(
+            _db, name, interval_seconds=interval_seconds,
+            default_start_time = self.VERY_LONG_AGO
+        )
+        self.batch_size = batch_size
+        #line 83-90 should be removable during refactoring
+        metadata_wrangler_url = Configuration.integration_url(
+                Configuration.METADATA_WRANGLER_INTEGRATION
+        )
+        if metadata_wrangler_url:
+            self.metadata_wrangler = SimplifiedOPDSLookup(metadata_wrangler_url)
+        else:
+            # This should only happen during a test.
+            self.metadata_wrangler = None
+        self.api = api or EnkiAPI.from_environment(self._db)
+        self.bibliographic_coverage_provider = (
+            EnkiBibliographicCoverageProvider(self._db, enki_api=api)
+        )
+
+    def run(self):
+        super(EnkiImport, self).run()
+
+    def run_once(self, start, cutoff):
+        # Give us five minutes of overlap because it's very important
+        # we don't miss anything.
+        since = start-self.FIVE_MINUTES
+        x=0
+        step=2000
+        while x < 80000:
```

Comment:
Why 80,000? Can you keep going until `self.api.availability` stops returning results?